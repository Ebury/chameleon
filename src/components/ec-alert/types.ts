export enum PropsAlertType {
  TEXT = 'text',
  DATE = 'date',
  NUMBER = 'number',
  TEL = 'tel'
}

export enum AlertType {
    ERROR = 'error', 
    INFO = 'info', 
    SUCCESS = 'success', 
    WARNING = 'warning'
}

export enum AlertEvent {
  UPDATE_OPEN = 'update:open',
  ACTION = 'action',
  CHANGE = 'change',
}

export interface AlertEvents {
  [AlertEvent.UPDATE_OPEN]: AlertProps['open']
  [AlertEvent.ACTION]: undefined
  [AlertEvent.CHANGE]: AlertProps['open']
}

export interface AlertProps {
  propsType?: PropsAlertType,
  type: AlertType,
  title: string,
  subtitle?: string,
  dismissable?: boolean,
  buttonText?: string,
  open?: boolean,
  responsive?: boolean
}
