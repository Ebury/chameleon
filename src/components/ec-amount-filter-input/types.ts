import type { DropdownItem } from '../ec-dropdown/types';

export interface ComparisonSymbolItem extends DropdownItem<string> {}

export interface AmountFilterInputModel {
  amount?: number,
  comparisonSymbol?: ComparisonSymbolItem,
}

export interface AmountFilterInputProps {
  modelValue: AmountFilterInputModel,
  locale?: string,
  isSensitive?: boolean,
  label?: string,
  note?: string,
  amountPlaceholder?: string,
  bottomNote?: string,
  isWarning?: boolean,
  warningTooltipMessage?: string,
  errorMessage?: string,
  errorTooltipMessage?: string,
  comparisonSymbolItems: ComparisonSymbolItem[],
}
