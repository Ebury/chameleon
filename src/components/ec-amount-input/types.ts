import type { InputFieldProps } from '../ec-input-field/types';

export interface AmountInputProps extends Omit<InputFieldProps, 'modelValue'> {
  modelValue?: number | string | null,
  isMasked?: boolean,
  locale?: string,
  currency?: string,
}
