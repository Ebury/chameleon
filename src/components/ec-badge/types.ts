import { AlertType as BadgeType } from '../ec-alert/types';

export { BadgeType };

export interface BadgeProps {
  type?: BadgeType,
  value: string | number,
}
