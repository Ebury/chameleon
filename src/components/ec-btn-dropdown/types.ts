import type { AnchorHTMLAttributes } from 'vue';
import type { RouteLocationRaw } from 'vue-router';

import type { DropdownItem } from '../ec-dropdown-search/types';

export interface BtnDropdownItem<TValue = string> extends DropdownItem<TValue> {
  href?: string,
  to?: string | RouteLocationRaw,
  attrs?: AnchorHTMLAttributes,
}

export interface BtnDropdownProps<TValue = string, TBtnDropdownItem extends BtnDropdownItem<TValue> = BtnDropdownItem<TValue>> {
  items?: TBtnDropdownItem[],
  isBtnDropdownDisabled?: boolean,
  buttonText: string,
  listDataTest?: string,
  href?: string,
  to?: string | RouteLocationRaw,
}
