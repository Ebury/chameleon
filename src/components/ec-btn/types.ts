import type { RouteLocationRaw } from 'vue-router';

import type { IconName } from '../ec-icon/iconNames';

export enum ButtonSize {
  Medium = 'md',
  Small = 'sm'
}

export enum ButtonCategory {
  Primary = 'primary',
  Secondary = 'secondary',
  Success = 'success',
  Error = 'error',
  Warning = 'warning'
}

export interface ButtonProps {
  size?: ButtonSize;
  href?: string;
  to?: RouteLocationRaw;
  tag?: string;
  isDisabled?: boolean;
  icon?: IconName;
  isRounded?: boolean;
  isOutline?: boolean;
  isFullWidth?: boolean;
  category: ButtonCategory;
  isReverse?: boolean;
  isLoading?: boolean;
  isSubmit?: boolean
}
