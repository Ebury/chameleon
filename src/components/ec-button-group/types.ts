export interface ButtonGroupItem<TValue> {
  id?: string | number,
  value: TValue,
  text: string,
  disabled?: boolean,
}

export interface ButtonGroupProps<TValue> {
  modelValue?: TValue,
  items: ButtonGroupItem<TValue>[],
}
