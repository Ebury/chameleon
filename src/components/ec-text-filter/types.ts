import type { InputFieldProps } from '../ec-input-field/types';

export interface TextFilterProps {
  modelValue?: InputFieldProps['modelValue']
  inputProps?: InputFieldProps
}

export enum TextFilterEvent {
  CHANGE = 'change',
}

export interface TextFilterEvents {
  [TextFilterEvent.CHANGE]: InputFieldProps['modelValue']
}
