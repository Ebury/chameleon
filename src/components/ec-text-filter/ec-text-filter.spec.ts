import type { ComponentMountingOptions } from '@vue/test-utils';
import { mount } from '@vue/test-utils';
import { defineComponent } from 'vue';

import type { CVueWrapper } from '../../../tests/utils/global';
import EcTextFilter from './ec-text-filter.vue';
import type { TextFilterProps } from './types';
import { TextFilterEvent } from './types';

describe('EcTextFilter', () => {
  function mountComponent(props?: TextFilterProps, mountOpts?: ComponentMountingOptions<TextFilterProps>): CVueWrapper {
    return mount<TextFilterProps>(
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      EcTextFilter as any,
      {
        props: {
          modelValue: 'Text test',
          ...props,
        },
        ...mountOpts,
      },
    ) as unknown as CVueWrapper;
  }

  function mountInputFieldAsTemplate(
    template: string,
    props: TextFilterProps,
    wrapperComponentOpts: Record<string, unknown>,
    mountOpts?: ComponentMountingOptions<TextFilterProps>,
  ) {
    const Component = defineComponent({
      components: { EcTextFilter },
      template,
      ...wrapperComponentOpts,
    });

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    return mount<TextFilterProps, ComponentMountingOptions<TextFilterProps>>(Component as any, {
      props,
      ...mountOpts,
    }) as unknown as CVueWrapper;
  }

  it('should display properly with the given props', () => {
    const wrapper = mountComponent();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should set the v-model on the value of the input and change when it changes', async () => {
    const wrapper = mountInputFieldAsTemplate(
      '<ec-text-filter v-model="text" />',
      {},
      {
        data() {
          return { text: '' };
        },
      },
    );

    expect((wrapper.findByDataTest('ec-input-field__input').element as HTMLInputElement).value).toBe('');
    await wrapper.setData({ text: 'some text' });
    expect((wrapper.findByDataTest('ec-input-field__input').element as HTMLInputElement).value).toBe('some text');
  });

  it('should render close icon depends on modelValue', async () => {
    const wrapper = mountInputFieldAsTemplate(
      '<ec-text-filter v-model="text" />',
      {},
      {
        data() {
          return { text: '' };
        },
      },
    );

    expect((wrapper.findByDataTest('ec-input-field__input').element as HTMLInputElement).value).toBe('');
    expect(wrapper.findByDataTest('ec-input-field__icon-wrapper').exists()).toBe(false);

    await wrapper.setData({ text: 'some text' });

    expect(wrapper.findByDataTest('ec-input-field__icon-wrapper')).toMatchSnapshot();
  });

  it('should emit the value when you write on the input', async () => {
    const wrapper = mountInputFieldAsTemplate(
      '<ec-text-filter v-model="text" />',
      {},
      {
        data() {
          return { text: '' };
        },
      },
    );

    expect((wrapper.findByDataTest('ec-input-field__input').element as HTMLInputElement).value).toBe('');
    await wrapper.findByDataTest('ec-input-field__input').setValue('some text');
    expect(wrapper.findComponent(EcTextFilter).emitted()[TextFilterEvent.CHANGE]?.[0]).toEqual(['some text']);
  });

  it('should emit the empty value when you click on close icon', async () => {
    const wrapper = mountInputFieldAsTemplate(
      '<ec-text-filter v-model="text" />',
      {},
      {
        data() {
          return { text: 'some text' };
        },
      },
    );

    await wrapper.findByDataTest('ec-input-field__icon').trigger('click');
    expect(wrapper.findComponent(EcTextFilter).emitted()[TextFilterEvent.CHANGE]?.[0]).toEqual(['']);
  });

  it('should render with a sensitive class when isSensitive prop is set to true', () => {
    const wrapper = mountComponent({ inputProps: { isSensitive: true } });

    expect(wrapper.element).toMatchSnapshot();
  });
});
