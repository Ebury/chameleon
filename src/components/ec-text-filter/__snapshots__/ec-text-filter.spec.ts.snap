// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EcTextFilter > should display properly with the given props 1`] = `
<div
  class="ec-text-filter"
  data-test="ec-text-filter"
>
  <div
    class="ec-input-field"
    data-test="ec-input-field"
  >
    <!--v-if-->
    <input
      class="ec-input-field__input ec-input-field__input--bg-gray-8 ec-input-field__input--has-icon ec-input-field__input--has-left-icon"
      data-test="ec-input-field__input ec-mock ec-tooltip-mock"
      id="ec-input-field-1"
      type="text"
    />
    <div
      class="ec-input-field__icon-wrapper ec-input-field__icon-wrapper--left"
      data-test="ec-input-field__left-icon-wrapper"
    >
      <svg
        class="ec-icon ec-input-field__icon"
        data-test="ec-input-field__left-icon"
        height="20"
        width="20"
      >
        <use
          xlink:href="#ec-simple-search"
        />
      </svg>
    </div>
    <div
      class="ec-input-field__icon-wrapper ec-input-field__icon-wrapper--right"
      data-test="ec-input-field__icon-wrapper"
    >
      <!--v-if-->
      <svg
        class="ec-icon ec-input-field__icon"
        data-test="ec-input-field__icon"
        height="20"
        width="20"
      >
        <use
          xlink:href="#ec-simple-close"
        />
      </svg>
    </div>
    <!--v-if-->
  </div>
</div>
`;

exports[`EcTextFilter > should render close icon depends on modelValue 1`] = `
<div
  class="ec-input-field__icon-wrapper ec-input-field__icon-wrapper--right"
  data-test="ec-input-field__icon-wrapper"
>
  <!--v-if-->
  <svg
    class="ec-icon ec-input-field__icon"
    data-test="ec-input-field__icon"
    height="20"
    width="20"
  >
    <use
      xlink:href="#ec-simple-close"
    />
  </svg>
</div>
`;

exports[`EcTextFilter > should render with a sensitive class when isSensitive prop is set to true 1`] = `
<div
  class="ec-text-filter"
  data-test="ec-text-filter"
>
  <div
    class="ec-input-field"
    data-test="ec-input-field"
  >
    <!--v-if-->
    <input
      class="ec-input-field__input ec-input-field__input--bg-gray-8 ec-input-field__input--has-icon ec-input-field__input--has-left-icon my-sensitive-content-test-class"
      data-test="ec-input-field__input ec-mock ec-tooltip-mock"
      id="ec-input-field-1"
      type="text"
    />
    <div
      class="ec-input-field__icon-wrapper ec-input-field__icon-wrapper--left"
      data-test="ec-input-field__left-icon-wrapper"
    >
      <svg
        class="ec-icon ec-input-field__icon"
        data-test="ec-input-field__left-icon"
        height="20"
        width="20"
      >
        <use
          xlink:href="#ec-simple-search"
        />
      </svg>
    </div>
    <div
      class="ec-input-field__icon-wrapper ec-input-field__icon-wrapper--right"
      data-test="ec-input-field__icon-wrapper"
    >
      <!--v-if-->
      <svg
        class="ec-icon ec-input-field__icon"
        data-test="ec-input-field__icon"
        height="20"
        width="20"
      >
        <use
          xlink:href="#ec-simple-close"
        />
      </svg>
    </div>
    <!--v-if-->
  </div>
</div>
`;
